share/eds/cache/accessor_cache.go

```diff
@@ -244,17 +248,31 @@ func newRefCloser(abs *accessorWithBlockstore) (*refCloser, error) {
 		return nil, err
 	}
 
-	var closeOnce sync.Once
-	return &refCloser{
+	rf := &refCloser{
 		accessorWithBlockstore: abs,
-		closeFn: func() {
-			closeOnce.Do(abs.removeRef)
-		},
-	}, nil
+		removeRef:              abs.removeRef,
+	}
+	// Set finalizer to ensure that accessor is closed when refCloser is garbage collected.
+	// We expect that refCloser will be closed explicitly by the caller. If it is not closed,
+	// we log an error.
+	runtime.SetFinalizer(rf, func(rf *refCloser) {
```

Comment:
How many such invocations we will have? Approx.

Reply:
few(1-10) per sec at most

Reply:
upto 100 per **second** ??

Reply:
Why I'm asking: `runtime.SetFinalizer` behaviour is easy to misuse, see docs (https://pkg.go.dev/runtime#SetFinalizer) & proposal to deprecated it (https://github.com/golang/go/issues/67535)

Adding up to 100/sec looks possible but better to bring attention to this line and solution in general, my 2c.

Reply:
This is one of 2 safety solutions we use for preventing hanging ref from blocking Accessor eviction from cache. 
- Finaliser closing forgotten ref.
- Force closing forgotten refs on timeout.

Given the fact, it is safety only mechanics and not main case. And assuming the only client of cache library is us, we can get away without Finaliser. Considering implications and concerns about Finaliser I'm inclined to remove it in favour of remaining forced closing only. 

@Wondertan thoughts?

Reply:
Cool